data/state/accountsDB.go

```diff
@@ -157,22 +156,18 @@ func (adb *AccountsDB) SaveDataTrie(accountHandler AccountHandler) error {
 	oldValues := make(map[string][]byte)
 
 	for k, v := range trackableDataTrie.DirtyData() {
-		originalValue := trackableDataTrie.OriginalValue([]byte(k))
+		flagHasDirtyData = true
 
-		if !bytes.Equal(v, originalValue) {
-			flagHasDirtyData = true
-
-			val, err := dataTrie.Get([]byte(k))
-			if err != nil {
-				return err
-			}
+		val, err := dataTrie.Get([]byte(k))
+		if err != nil {
+			return err
+		}
 
-			oldValues[k] = val
+		oldValues[k] = val
 
-			err = dataTrie.Update([]byte(k), v)
-			if err != nil {
-				return err
-			}
+		err = dataTrie.Update([]byte(k), v)
```

Comment:
that was with problems, and it was wrong. As originalData was always nil, so any new change would have been written. In case of delete, the save would not work and nil would be equal to nil.

Reply:
The if branch on line 162 prevented deleting a (key, value) when the account would have been refetched from db.